plugins/engines/optix/OptiXScene.cpp

```diff
@@ -524,4 +523,127 @@ void OptiXScene::commitSimulationData()
+    if( !volumeHandler )
+    {
+        _volumeBuffer = _context->createBuffer( RT_BUFFER_INPUT, RT_FORMAT_UNSIGNED_BYTE, 0 );
+        _context[ "volumeData" ]->setBuffer( _volumeBuffer );
+        _context[ "volumeDimensions" ]->setUint( 0, 0, 0 );
+        _context[ "volumeScale" ]->setFloat( 0.f, 0.f, 0.f );
+        _context[ "volumePosition" ]->setFloat( 0.f, 0.f, 0.f );
+        _context[ "volumeEpsilon" ]->setFloat( 0.f );
+        _context[ "volumeDiag" ]->setFloat( 0.f );
+        return;
+    }
+
+    if( _volumeBuffer == 0 )
+    {
+        _volumeBuffer = _context->createBuffer(
+            RT_BUFFER_INPUT, RT_FORMAT_UNSIGNED_BYTE, volumeHandler->getSize() );
+        uint64_t size = volumeHandler->getSize() * sizeof( unsigned char );
+        if( size != 0 )
+        {
+            memcpy( _volumeBuffer->map(), volumeHandler->getData(), size );
+            _volumeBuffer->unmap();
+        }
+        _context[ "volumeData"   ]->setBuffer( _volumeBuffer );
+
+        // Optix needs a bounding box around the volume so that if can find intersections
+        // before initiating the traversing of the volume
+        _buildVolumeAABBGeometry();
+    }
+
+    const Vector3ui& dimensions = volumeHandler->getDimensions();
+    _context[ "volumeDimensions" ]->setUint( dimensions.x(), dimensions.y(), dimensions.z());
+
+    VolumeParameters& volumeParameters = _parametersManager.getVolumeParameters();
+    const Vector3f& scale = volumeParameters.getScale();
+    _context[ "volumeScale" ]->setFloat( scale.x(), scale.y(), scale.z());
+
+    const Vector3f& position = volumeParameters.getPosition();
+    _context[ "volumePosition" ]->setFloat( position.x(), position.y(), position.z());
+
+    _context[ "volumeEpsilon" ]->setFloat(
+        volumeHandler->getEpsilon( scale, volumeParameters.getSamplesPerRay()));
+
+    Vector3f diag = Vector3f( dimensions ) * scale;
+    const float volumeDiag = std::max( diag.x(), std::max( diag.y(), diag.z()));
+    _context[ "volumeDiag" ]->setFloat( volumeDiag );
+}
+
+void OptiXScene::commitTransferFunctionData()
+{
+    if( _colorMapBuffer == 0 )
+        _colorMapBuffer = _context->createBuffer(
+            RT_BUFFER_INPUT, RT_FORMAT_FLOAT4, _transferFunction.getDiffuseColors().size() );
+
+    uint64_t size = _transferFunction.getDiffuseColors().size() * 4 * sizeof( float );
+    if( size != 0 )
+    {
+        memcpy( _colorMapBuffer->map(), &_transferFunction.getDiffuseColors()[0], size );
+        _colorMapBuffer->unmap();
+    }
+
+    _context[ "colorMap" ]->setBuffer( _colorMapBuffer );
+    _context[ "colorMapSize" ]->setUint( _transferFunction.getDiffuseColors().size() );
+    _context[ "colorMapMinValue" ]->setFloat( _transferFunction.getValuesRange().x() );
+    _context[ "colorMapRange" ]->setFloat(
+        _transferFunction.getValuesRange().y() - _transferFunction.getValuesRange().x() );
+
+}
+
+void OptiXScene::_buildVolumeAABBGeometry()
```

Comment:
They could but since they address 2 very different topics, that would need too many parameters.